Code Review

radavis opened the pull request:
Do not merge this branch!

This branch was created and reset to a later commit in order to conduct a code review.

diegodesouza committed:
Integration tests, factory girl
create a listing successfully
tests passings

diegodesouza committed:
Merge branch 'master' of github.com:diegodesouza/Places

Conflicts:
	spec/features/places/user_creates_a_listing_spec.rb
	spec/features/places/user_edits_a_listing_spec.rb
	spec/support/factory_girl.rb

diegodesouza committed:
user creates listings successfully
test for user creating a listing passing

diegodesouza committed:
user edits own listing successfully

diegodesouza committed:
user views all listings
test passing

diegodesouza committed:
user is able to see a specific listing

diegodesouza committed:
user deletes a listing
test passing

diegodesouza committed:
fixed validation for edit and delete, restricting to current user
all testing passing

diegodesouza committed:
fixed edit and delete button validation when signout

diegodesouza committed:
create reservation form
reservation model
reservation controller

diegodesouza committed:
date being stored in the database

diegodesouza committed:
reservation being displaied in the show page

diegodesouza committed:
added some styling
create user show page

diegodesouza committed:
carrierwave
fog
imageuploader

diegodesouza committed:
carrierwave::minimagick
photo uploading working

diegodesouza committed:
some styling done on forms

diegodesouza committed:
add procfile

diegodesouza committed:
fixed deprecation warning config.serve_static_assets from config.serve_static_files

diegodesouza committed:
unicorn config

diegodesouza committed:
vendor compiled assets

diegodesouza committed:
environments test.rb

diegodesouza committed:
debugging

diegodesouza committed:
debugging

diegodesouza committed:
debugging part 3

diegodesouza committed:
trying the images to heroku

diegodesouza committed:
fixing image for heroku

diegodesouza committed:
fixed margin for sign_in/sign_up form

diegodesouza committed:
Before working with spencer

diegodesouza committed:
reverted back to no css
will implement css little by little.

diegodesouza committed:
Merge pull request #1 from diegodesouza/spencer-refactoring

reverted back to no css

diegodesouza committed:
images working, diploying to heroku with images

diegodesouza committed:
Merge pull request #2 from diegodesouza/listing-availability-by-date

images working, diploying to heroku with images

diegodesouza committed:
working on calendar availability feature.

diegodesouza committed:
Merge pull request #3 from diegodesouza/listing-availability-by-date

working on calendar availability feature.

diegodesouza committed:
some styling

diegodesouza committed:
Merge pull request #4 from diegodesouza/listing-availability-by-date

some styling

diegodesouza committed:
syntax error off css when deploying to heroku

diegodesouza committed:
Merge pull request #5 from diegodesouza/listing-availability-by-date

syntax error off css when deploying to heroku

diegodesouza committed:
added somestyling

diegodesouza committed:
fixed http:// for fonts and ajax, javascript in order to push changes to heroku
fixex typo

diegodesouza committed:
fixed http:// for fonts and ajax, javascript in order to push changes to heroku

diegodesouza committed:
Merge pull request #6 from diegodesouza/listing-availability-by-date

Listing availability by date

diegodesouza committed:
Merge branch 'master' of github.com:diegodesouza/Places into listing-availability-by-date

diegodesouza committed:
Commiting this in the listing-availability-by-date by mistake

diegodesouza committed:
limiting users to make a reservation to a listing when it's unavailable

diegodesouza committed:
Merge pull request #7 from diegodesouza/listing-availability-by-date

Listing availability by date

diegodesouza committed:
fixed schema and added id to current user in controller

diegodesouza committed:
Merge pull request #8 from diegodesouza/fix-schema

fixed schema and added id to current user in controller

diegodesouza committed:
created reviews controller
created reviews model
created reviews views
reviews working partially
will test soon

diegodesouza committed:
Merge pull request #9 from diegodesouza/reviews

created reviews controller

diegodesouza committed:
user can make reservations successfully
user can leave a review on a reservation he made successfully

diegodesouza committed:
Merge pull request #10 from diegodesouza/reviews

user can make reservations successfully

diegodesouza committed:
styling on show page for reservation

diegodesouza committed:
added some styling
creates reservation successfully
reservation test not passing yet

diegodesouza committed:
Merge pull request #11 from diegodesouza/show-styling

Show styling

diegodesouza committed:
user edits own review test passing

diegodesouza committed:
user gets an error if left fields blank
all tests passing

diegodesouza committed:
user deletes own review successfully

diegodesouza committed:
Merge pull request #12 from diegodesouza/user_edits_deletes_reviews

User edits deletes reviews

diegodesouza committed:
little styling on the show page review section

diegodesouza committed:
Merge pull request #13 from diegodesouza/some-styling

little styling on the show page review section

diegodesouza committed:
user searches listings by title, city, state, property_type successfully
test passing

diegodesouza committed:
Merge pull request #14 from diegodesouza/user-searches-listings

user searches listings by title, city, state, property_type successfully

diegodesouza committed:
created pagination
will fix the link floating to left, for when listings is less than
8 per page

diegodesouza committed:
Merge pull request #15 from diegodesouza/listings-pagination

created pagination

diegodesouza committed:
map in show page working

diegodesouza committed:
Merge pull request #16 from diegodesouza/maps

map in show page working

diegodesouza committed:
styling to search form

diegodesouza committed:
Merge pull request #17 from diegodesouza/styling

styling to search form

diegodesouza committed:
added config.assets.initialize_on_precompile = false to application.rb

diegodesouza committed:
Merge pull request #18 from diegodesouza/fix-to-deploy-to-heroku

added config.assets.initialize_on_precompile = false to application.rb

diegodesouza committed:
removed ajax/jquery from application.js

diegodesouza committed:
Merge pull request #19 from diegodesouza/fix-to-deploy-to-heroku

removed ajax/jquery from application.js

diegodesouza committed:
fixed search test to compensate css that was done

diegodesouza committed:
Merge pull request #20 from diegodesouza/fix-search-test

fixed search test to compensate css that was done

diegodesouza committed:
tests passing
added navbar styling

diegodesouza committed:
Merge pull request #21 from diegodesouza/navbar-styling

tests passing

diegodesouza committed:
fixed navbar that was pushing the content down
some styling to profile page

diegodesouza committed:
added styling to profile page

diegodesouza committed:
Merge pull request #22 from diegodesouza/styling-profile-page

Styling profile page

diegodesouza committed:
styled forms
styled show a little more
styled buttons
fixed validations

diegodesouza committed:
Merge pull request #23 from diegodesouza/style-reservation-show

styled forms

diegodesouza committed:
some more changes, moved css to some specific files

diegodesouza committed:
Update README.rdoc

diegodesouza committed:
made changes to show page, index, flash notices
created images model for uploading in the show

diegodesouza committed:
Merge pull request #24 from diegodesouza/styling-changes

Styling changes

diegodesouza committed:
created uploaders for more images
added slick carousel index and show page
removed kaminari pagination

diegodesouza committed:
changed back to pagination for index page
some styling done

diegodesouza committed:
changed back to pagination for index page
some styling done

diegodesouza committed:
Merge pull request #25 from diegodesouza/styling-changes

Styling changes

diegodesouza committed:
Merge branch 'master' of github.com:diegodesouza/Places into styling-changes

Conflicts:
	app/assets/stylesheets/slick.css
	app/assets/stylesheets/slick/slick.css
	app/assets/stylesheets/slick/slick.css.scss

diegodesouza committed:
Merge pull request #26 from diegodesouza/styling-changes

Styling changes

diegodesouza committed:
fixed typo

diegodesouza committed:
Merge pull request #27 from diegodesouza/styling-changes

fixed typo

diegodesouza committed:
fixed require_tree ./slick in application.css.scss

diegodesouza committed:
Merge pull request #28 from diegodesouza/styling-changes

fixed require_tree ./slick in application.css.scss

radavis commented on app/controllers/application_controller.rb:
```diff
@@ -1,5 +1,20 @@
+  end
+
+  def authorized
+    unless current_user.id == Reservation.find(params[:listing_id]).user
```
This method will throw an error fail if `current_user` is nil. Also you are comparing an integer to an object. 

It seems like you want to have a helper method that redirects users who don't own a Reservation. 

```
reservation = Reservation.find(params[:reservation_id])
unless current_user && reservation && current_user == reservation.user
  redirect_to root_path, notice: "You are not authorized to view this Reservation"
end
```

radavis commented on app/controllers/homes_controller.rb:
```diff
@@ -1,4 +1,2 @@
 class HomesController < ApplicationController
-  def index
-  end
 end
```
You can delete this controller since you are not referencing it in your routes.

radavis commented on app/controllers/reviews_controller.rb:
```diff
@@ -0,0 +1,48 @@
+    @listing = Listing.find(params[:listing_id])
+    @review = Review.new(review_params)
+    @review.listing_id = @listing.id
+    @review.user_id = current_user.id
```
Always use the active record object instead of the id.

```
@review.listing = @listing
@review.user = current_user
```

radavis commented on app/helpers/states_helper.rb:
```diff
@@ -0,0 +1,59 @@
+      ['Wyoming', 'Wyoming']
+    ]
+  end
+end
```
Is there any particular reason why this is a nested array?

diegodesouza replied:
couldn't get it working otherwise, tried just the arrays 
['Wisconsin', 'Wisconsin']
['Wyoming', 'Wyoming']
but it wouldnt show on my drop down.

radavis replied:
Instead of using the `select` method in your views, like so:

```
<%= f.select(:state, options_for_select(us_states)) %>
```

you could change your us_states method to return an array of states, and use the collection_select method:

```
<%= f.collection_select :state, us_states, :to_s, :to_s %>
```

radavis commented on app/models/reservation.rb:
```diff
@@ -0,0 +1,28 @@
+    if check_in.nil?
+      errors.add(:check_in, "Check in can't be blank")
+    elsif check_out.nil?
+      errors.add(:check_out, "Check out can't be blank")
```
Use the presence validators for check_in and check_out.

radavis commented on app/uploaders/image_four_uploader.rb:
```diff
@@ -0,0 +1,27 @@
+  version :thumb do
+    process :resize_to_fill => [200, 200]
+  end
+end
```
If all of these image uploaders are the same, create a base uploader and inherit from it.

diegodesouza replied:
do you mean like a helper? can you give me an example of what it would look like?

radavis replied:
Inheritance allows us to reuse methods from another class. For instance, when one of our models inherits from ActiveRecord, we gain access to the `.all` and `.where` methods.

I noticed that all of your image uploader classes looked very similar. Instead of duplicating code, you can create one `image_uploader.rb` class and then inherit from it.

```
class ImageFourUploader < ImageUploader
end
```

radavis commented on app/views/devise/passwords/new.html.erb:
```diff
@@ -9,8 +10,9 @@
 
   <div class="actions">
-    <%= f.submit "Send me reset password instructions" %>
+    <%= f.submit "Send me reset password instructions", class:"this-thing" %>
```
"this-thing" isn't a very descriptive class name. "submit-button" is better.

diegodesouza replied:
agree, i was mad at it because it wouldn't override the foundation css. so I just called it that lol, and it magically worked

diegodesouza replied:
also the css there applies to all my buttons so Id have to call it a more general button, this-button? lol

radavis commented on app/views/layouts/application.html.erb:
```diff
@@ -2,44 +2,51 @@
+    <%= javascript_tag do %>
     $(function(){
       $(document).foundation();
     });
```
```
<%= javascript_tag do %>
    $(function(){
      $(document).foundation();
    });
<% end %>
```

This block is in your application.js file, so you can delete it here. This is an issue with the make_it_so gem.

radavis commented on app/views/layouts/application.html.erb:
```diff
@@ -2,44 +2,51 @@
+    <%= yield :extra_footer %>
+
+  </div>
+<script type="text/javascript" src="//cdn.jsdelivr.net/jquery.slick/1.3.15/slick.min.js"></script>
```
You can use the rails javascript include tag helper here:

```
<%= javascript_include_tag "//cdn.jsdelivr.net/jquery.slick/1.3.15/slick.min.js" %>
```

Note that some content delivery services (CDNs) will not serve assets to heroku domains. You may want to copy this file to your `app/assets/javascripts` folder.

radavis commented on app/views/listings/show.html.erb:
```diff
@@ -0,0 +1,78 @@
+      </p>
+
+      <% if current_user == nil %>
+      <% elsif @listing.user == current_user %>
```
`if current_user == @listing.user`

radavis commented on app/views/listings/show.html.erb:
```diff
@@ -0,0 +1,78 @@
+        Country: <%= @listing.country %><br><hr>
+      </p>
+
+      <% if current_user == nil %>
```
Delete this line

radavis commented on config/routes.rb:
```diff
@@ -1,58 +1,15 @@
-  #     # (app/controllers/admin/products_controller.rb)
-  #     resources :products
-  #   end
+  resources :listings do
```
If you only want a route to the listings controller show action, you will need to disable some routes here:

```
resources :listings, only: [] do
  ...
end
```

radavis commented on spec/factories_spec.rb:
```diff
@@ -0,0 +1,10 @@
+#       expect(FactoryGirl.build(factory)).to be_valid
+#     end
+#   end
+# end
```
If you aren't using this, you can delete it.

radavis commented on spec/features/reservations/user_makes_a_reservation_spec.rb:
```diff
@@ -0,0 +1,62 @@
+  #   expect(page).to_not have_content "January 01 2015 at 1:00 am"
+  #   expect(page).to have_content "Dates are not availble at this time"
+  # end
+end
```
Instead of commenting an entire test, you can either mark it pending or disable it:

```
scenario "test for something" do
pending "this test will not run"

end

xscenario "test something else" do
  # will not run
end
```

diegodesouza committed:
minor fixes

diegodesouza committed:
code review changes.

diegodesouza committed:
Merge pull request #30 from diegodesouza/codereview

code review changes.

diegodesouza committed:
changed timeout unicorn to 30

diegodesouza committed:
changed timeout unicorn to 30

diegodesouza committed:
Merge branch 'master' of github.com:diegodesouza/Places

diegodesouza committed:
trying to fix slick to be deployed

diegodesouza committed:
still getting syntax error

diegodesouza committed:
updated readme and typo with slick

diegodesouza committed:
updated readme and typo with slick

diegodesouza committed:
Merge branch 'master' of github.com:diegodesouza/Places

Conflicts:
	README.rdoc

diegodesouza committed:
deleted some images'

diegodesouza committed:
fix bug

diegodesouza committed:
Listings search through AJAX successfully.

diegodesouza committed:
Pagination working with AJAX

diegodesouza committed:
removed gem foundation datepicker

diegodesouza committed:
Refactor controller a bit. needs more work

diegodesouza committed:
Fix some of the error messages.

diegodesouza committed:
Fix the reservation partial to not show to listing owner.

diegodesouza committed:
Fix forms stylying a little, needs more work.

diegodesouza committed:
Refactor routes

diegodesouza committed:
After importing the stylesheets:
  * Navigation bar/dropdown for mobile working.

diegodesouza committed:
Dropdown works for larger screens.
Removes media ranges.

diegodesouza committed:
removed some unused files, made the hamburger reflect the origial design.

diegodesouza committed:
Fix the nav-bar for the moble version. Where it pushs downt the content
to have a little more room for touching.
Fix it for larger screens also.

diegodesouza committed:
Title for small screens done.

diegodesouza committed:
Have the search bar working to some extent.

Title is full viewport height for small screens

diegodesouza committed:
Image and search bar matches original design.

diegodesouza committed:
Fix section title

remove unused files

diegodesouza committed:
Adds styling for listings in the index page.

diegodesouza committed:
Fix the index page footer.

diegodesouza committed:
Fix stying for index page.

diegodesouza committed:
Deletes unused files

fix the show page.

diegodesouza committed:
Fix show page for larger screens.

diegodesouza committed:
Fix the new listing page.

diegodesouza committed:
Form effect not working anymore.

diegodesouza committed:
Form effect working.

diegodesouza committed:
styling the button in the form

diegodesouza committed:
Fix index page image.

diegodesouza committed:
Fix profile page.

diegodesouza committed:
Fix button in editing listing.

diegodesouza committed:
Fix footer styling.

diegodesouza committed:
Sign up page styled.

diegodesouza committed:
styled all devise forms to reflect the design.

diegodesouza committed:
Changed the gem file to use minitest instead.

diegodesouza committed:
Set up test enviroment with mini test

Fix date picker positioning on the page, needs more styling.

diegodesouza committed:
Styled datepicker.

diegodesouza committed:
Styling the review page/forms

diegodesouza committed:
Fix datepicker on larger screens

diegodesouza committed:
Some styling for the footer.

diegodesouza committed:
fix the footer styling.

diegodesouza committed:
Fix the paginator through remote true ajax

diegodesouza committed:
Removed masonry.

diegodesouza committed:
Fix the image, adds a slider in the index page and show page.

diegodesouza committed:
Fix date picker range, when the user selects the check in, and goes to

select the checkout date, he wont be able to checkout on the same day

and the restrict dates that have been selected.

diegodesouza committed:
Fix image in the index page.

diegodesouza committed:
delete unused files, and unused methods.

diegodesouza committed:
Merge pull request #31 from diegodesouza/styling-from-scratch

Styling from scratch

diegodesouza committed:
Add config.fog_provider to carrierwave config.

diegodesouza committed:
Remove fog configuration, trying to get images to show.

diegodesouza committed:
Remove fog configuration, trying to get images to show.

diegodesouza committed:
Merge branch 'master' of github.com:diegodesouza/Places

diegodesouza committed:
Merge branch 'master' of github.com:diegodesouza/Places

diegodesouza committed:
Merge branch 'master' of github.com:diegodesouza/Places

diegodesouza committed:
Attempt to fix the s3 bug.

diegodesouza committed:
Fix aws s3 credentials.

diegodesouza committed:
Change the gem to fog-aws change the carrierwave config to what was before. Remove description validation for max characters in description.

diegodesouza committed:
Fix making a reservation, removed extra code in routes. fix the profile show page to show the reservations.

diegodesouza committed:
Move render heading to layout.

diegodesouza committed:
Displaying form error messages.

diegodesouza committed:
position relative for background image.

diegodesouza committed:
position relative for background image.

diegodesouza committed:
Merge branch 'master' of https://git.heroku.com/places-launch

diegodesouza committed:
Change image to max width and height auto.

diegodesouza committed:
Change image to max width and height auto.

diegodesouza committed:
Merge branch 'master' of https://git.heroku.com/places-launch

diegodesouza committed:
Merge branch 'master' of https://git.heroku.com/places-launch

diegodesouza committed:
Merge branch 'master' of https://git.heroku.com/places-launch

diegodesouza committed:
Merge branch 'master' of https://git.heroku.com/places-launch

diegodesouza committed:
add config.serve_static_assets = true so that heroku can serve my image from the app/code.

diegodesouza committed:
Try with a small image.

diegodesouza committed:
rever back to reg image. TODO: figure out why image is not scaling after pushing to heroku.

diegodesouza committed:
.

diegodesouza committed:
precompile to attempt to have the image flow.

diegodesouza committed:
add parent container to parallax.

diegodesouza committed:
no need for config.serve_static_files = true.

diegodesouza committed:
adds a cropped image for mobile

diegodesouza committed:
another attempt to fix the production image.

diegodesouza committed:
another attempt to fix the production image.

diegodesouza committed:
Merge branch 'master' of https://git.heroku.com/places-launch

diegodesouza committed:
minor changes to get proper image for mobile in production.

diegodesouza committed:
precompile assets.

diegodesouza committed:
minor changes to get proper image for mobile in production.

diegodesouza committed:
another try

diegodesouza committed:
precompile assets.

diegodesouza committed:
testing new image size on mobile.

diegodesouza committed:
Fix google maps.

diegodesouza committed:
rake assets:clobber

diegodesouza committed:
rake assets:clobber

diegodesouza committed:
Merge branch 'master' of https://github.com/diegodesouza/Places

diegodesouza committed:
Fix typo

diegodesouza committed:
Trying another image for mobile.

diegodesouza committed:
config/production.rb config.assets.compile = true

diegodesouza committed:
precompile it locally.

diegodesouza committed:
another try.

diegodesouza committed:
.

diegodesouza committed:
precompile.

diegodesouza committed:
.

diegodesouza committed:
fix some styling and issues with error messages.

diegodesouza committed:
Removed image from mobile view.

diegodesouza committed:
fix media query.

diegodesouza committed:
fix border radius in button.
